plugins/providers/bigquery/client.go

```diff
@@ -327,6 +306,34 @@ func (c *bigQueryClient) ListAccess(ctx context.Context, resources []*domain.Res
 	return access, nil
 }
 
+func (c *bigQueryClient) getGrantAbleRoleForDataset() ([]bq.AccessRole, error) {
+	var roles = []bq.AccessRole{bq.OwnerRole, bq.WriterRole, bq.ReaderRole}
+	var resourceName string
+	ctx := context.Background()
+	datasetIterator := c.client.Datasets(ctx)
+	for {
+		dataset, err := datasetIterator.Next()
```

Comment:
Would this iterate over each dataset and their roles in a given project while validating the given role in the config?

If yes... this iteration will take too much time - if a given project has many datasets.

Reply:
nope, it'll iterate only one dataset and break, There is no way to add custom role on bigquery dataset so every dataset will have same sort of roles